### PR TITLE
Supports class level setUp and Teardown

### PR DESCRIPTION
**Pull request by steveli777**
To support class level setUp and Teardown for Unittest

### CODE REVIEW

**Comment by andrewleech** [comment]
Hi @steveli777 thanks for the contribution! As a general rule (not yet documented anywhere easy to find) the commits here should follow a pattern of `module: change description`, for example take a look at the commits in https://github.com/micropython/micropython-lib/pull/488/commits

File: python-stdlib/unittest/unittest.py [hunk 1]
```diff
@@ -188,8 +188,14 @@ def wasSuccessful(self):
 # TODO: Uncompliant
 def run_class(c, test_result):
     o = c()
+    set_up_class = getattr(o, "setUpClass", lambda: None)
+    tear_down_class = getattr(o, "tearDownClass", lambda: None)
     set_up = getattr(o, "setUp", lambda: None)
     tear_down = getattr(o, "tearDown", lambda: None)
+
+    if set_up_class:
+        set_up_class()
```
Comment (andrewleech) [inline python-stdlib/unittest/unittest.py:197]: In the `getattr` line above, if the function doesn't exist it defaults to `lambda: None`. This is done so that you don't need the `if set_up_class` here, you can just run `set_up_class()` and if it doesn't exist it'll just run the no-op lambda function

File: python-stdlib/unittest/unittest.py [hunk 2]
```diff
@@ -198,9 +204,9 @@ def run_class(c, test_result):
             try:
                 test_result.testsRun += 1
                 m()
-                print(" ok")
+                print(" PASS")
             except SkipTest as e:
-                print(" skipped:", e.args[0])
+                print(" SKIPPED:", e.args[0])
```
Comment (andrewleech) [inline python-stdlib/unittest/unittest.py:209]: These output format changes should be done in a separate commit - but only if the new format matches cpython unittest? I'd prefer to add this commit after the #488 changeset which has had some other changes made to the output format already to more closely match cpython.

File: python-stdlib/unittest/unittest.py (hunk 1 above)
Comment (steveli777) [reply python-stdlib/unittest/unittest.py:197]: Thanks for the suggestion, fixed.

File: python-stdlib/unittest/unittest.py (hunk 2 above)
Comment (steveli777) [reply python-stdlib/unittest/unittest.py:209]: You are right, I reverted those text changes.

**Review by mattytrentini — APPROVED** [review]
Clean and simple, nice!

**Comment by andrewleech** [comment]
Hi @steveli777 as I've got a running unittest PR on the go, I've rebased your commit here on top of that one. 

It needed some manual intervention to deal with merge conflicts, but I think I got it right: https://github.com/micropython/micropython-lib/pull/488/commits/23866f3848e3f15878d0a69954268a81d96e4b1f

Even with the merge changes the commit is still comitted in your name, I hope that's good for you!

If you get a chance could you review / test this setUpClass functionality in the branch from https://github.com/micropython/micropython-lib/pull/488 ?

Otherwise do you have a simple test case I can run (or add as unit test) to cover this functionality?

**Comment by dpgeorge** [comment]
This feature was implemented in 2d61dbdb93fa941c54b3990c2ee046074d2fce07
